packages/forms-web-app/__tests__/unit/controllers/examination/check-your-answers/utils/summary-list-item/submitting-for.test.js

```diff
@@ -0,0 +1,78 @@
+const {
+	getSummaryListItemSubmittingFor
+} = require('../../../../../../../src/controllers/examination/check-your-answers/utils/summary-list-item/submitting-for');
+
+const {
+	getDeadlineDetailsSubmittingFor
+} = require('../../../../../../../src/controllers/examination/session/deadline');
+const {
+	getSummaryListItem
+} = require('../../../../../../../src/controllers/utils/get-summary-list-item');
+const {
+	getSelectedOptionText
+} = require('../../../../../../../src/controllers/examination/check-your-answers/utils/summary-list-item/helpers');
+
+jest.mock('../../../../../../../src/controllers/examination/session/deadline', () => ({
+	getDeadlineDetailsSubmittingFor: jest.fn()
+}));
+jest.mock('../../../../../../../src/controllers/utils/get-summary-list-item', () => ({
+	getSummaryListItem: jest.fn()
+}));
+jest.mock(
+	'../../../../../../../src/controllers/examination/check-your-answers/utils/summary-list-item/helpers',
+	() => ({
+		getSelectedOptionText: jest.fn()
+	})
+);
+
+describe('controllers/examination/check-your-answers/utils/summary-list-item/submitting-for', () => {
+	describe('#getSummaryListItemSubmittingFor', () => {
+		const req = {
+			session: { mockSession: 'mock session' }
+		};
+		describe('When getting the submitting for summary list item for the check your answers page', () => {
+			describe('and the submitting for value retrived from the session is NOT myself, organisation or agent', () => {
+				it('should throw an error', () => {
+					expect(() => getSummaryListItemSubmittingFor(req.session)).toThrowError(
+						'Submitting for text is undefined'
+					);
+				});
+			});
+			describe('and the submitting for value retrived from the session is myself, organisation or agent', () => {
+				let result;
+				const mockSubmittingFor = 'myself/organisation/agent';
+				const mockSubmittingForText = 'mock submitting for text';
+				const mockSummaryListItem = {
+					mockSummaryListItem: 'mock summary list item'
+				};
+				beforeEach(() => {
+					getDeadlineDetailsSubmittingFor.mockReturnValue(mockSubmittingFor);
+					getSelectedOptionText.mockReturnValue(mockSubmittingForText);
+					getSummaryListItem.mockReturnValue(mockSummaryListItem);
+					result = getSummaryListItemSubmittingFor(req.session);
+				});
+				it('should call the functions', () => {
+					expect(getDeadlineDetailsSubmittingFor).toHaveBeenCalledWith(req.session);
+					expect(getSelectedOptionText).toHaveBeenCalledWith(
+						{
```

Comment:
This seems like a lot of logic to be nested with others.

If this is using config then lift into it's own `it` and say what is needs / is doing